app/lib/document-loader/index.js

```diff
@@ -0,0 +1,47 @@
+const { RecursiveCharacterTextSplitter } = require('langchain/text_splitter')
+
+const { DOC, PDF, DOCX, TXT } = require('../../constants/document-types')
+const loadDocx = require('./docx')
+const loadPdf = require('./pdf')
+const loadText = require('./txt')
+
+const loaders = {
+  [DOCX]: loadDocx,
+  [DOC]: loadDocx,
+  [PDF]: loadPdf,
+  [TXT]: loadText
+}
+
+const loadDocument = async (document) => {
+  const blobMetadata = document.metadata
+
+  const loader = loaders[document.contentType]
+
+  if (!loader) {
+    throw new Error(`Unsupported document type: ${document.contentType}`)
+  }
+
+  const docs = await loader(document)
+
+  const docsWithMetadata = docs.map(doc => ({
+    ...doc,
+    metadata: {
+      ...doc.metadata,
+      blobMetadata: {
+        ...blobMetadata,
+        contentType: document.contentType
+      }
+    }
+  }))
+
+  const splitter = new RecursiveCharacterTextSplitter({
```

Comment:
Not sure if we need this here. It will be better just to return the documents as is and split (if required) in the calling service.

Reply:
Good shout, I lifted this from knowledge which uses splitting for vector creation, this was basically splitting then rejoining straight away... new push removes this